x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/DateUtils.java

```diff
@@ -136,8 +136,14 @@ public static String toString(Object value) {
             sb.append(":");
             durationInSec = durationInSec % SECONDS_PER_MINUTE;
             sb.append(indent(durationInSec));
-            sb.append(".");
-            sb.append(TimeUnit.NANOSECONDS.toMillis(d.getNano()));
+            long millis = TimeUnit.NANOSECONDS.toMillis(d.getNano());
+            if (millis > 0) {
+                sb.append(".");
+                while (millis % 10 == 0) {
```

Comment:
Alternatively you could just iterate on the string millis from end->start and remove all zeroes.
Don't know if performance wise makes sense...

Reply:
I've also fluttered a bit between the two approaches. I went with this approach since generally arithmetic operations should be faster. But anyways, micro-optimisations.